### DISCUSSION TITLE
Understanding ParseFile

### DISCUSSION THREAD

**Discussion by proj-sashido** [netreconlab/Parse-Swift#107] [Q&A] [answered]
This is more just a sanity check than a request for help.

When using the `fetch()` function, what extra information is added to the file?
Am I correct that only one property changes, and that is `localURL`?

I was momentarily confused as I was trying to figure out why `file.data` was consistently nil even after fetching, but I’m assuming it simply saves the file locally and allows you to access it from its `localURL`.

**Comment by cbaker6** [comment, marked as answer]
Yes, it fetches/downloads the file from your Parse Server, AWS S3, or other bucket and stores it at `localURL`. The [documentation](https://swiftpackageindex.com/netreconlab/parse-swift/5.6.0/documentation/parseswift/parsefile) and playgrounds describe the behavior: https://github.com/netreconlab/Parse-Swift/blob/1a238600b716b0a111d59865f2bf62f46e973cbd/ParseSwift.playground/Pages/9%20-%20Files.xcplaygroundpage/Contents.swift#L178-L182

**Comment by proj-sashido** [comment]
Thank you for clarifying :)
